bundler/lib/bundler/rubygems_integration.rb

```diff
@@ -119,7 +119,7 @@ def inflate(obj)
     def correct_for_windows_path(path)
       if Gem::Util.respond_to?(:correct_for_windows_path)
         Gem::Util.correct_for_windows_path(path)
-      elsif path[0].chr == "/" && path[1].chr =~ /[a-z]/i && path[2].chr == ":"
+      elsif path[0].chr == "/" && path[1].chr.match?(/[a-z]/i) && path[2].chr == ":"
```

Comment:
I'm planning to delete this code at #7258, which suggests that we can probably apply the same fix in the rubygems side.